crates/ty_ide/src/completion.rs

```diff
@@ -5872,15 +5862,169 @@ from .imp<CURSOR>
             .auto_import()
             .module_names();
         assert_snapshot!(builder.build().snapshot(), @r"
-        Deprecated :: importlib.metadata
-        DeprecatedList :: importlib.metadata
-        DeprecatedNonAbstract :: importlib.metadata
-        DeprecatedTuple :: importlib.metadata
         deprecated :: typing_extensions
         deprecated :: warnings
         ");
     }
 
+    #[test]
+    fn reexport_simple_import_noauto() {
+        let snapshot = CursorTest::builder()
+            .source(
+                "main.py",
+                r#"
+import foo
+foo.ZQ<CURSOR>
+"#,
+            )
+            .source("foo.py", r#"from bar import ZQZQ"#)
+            .source("bar.py", r#"ZQZQ = 1"#)
+            .completion_test_builder()
+            .module_names()
+            .build()
+            .snapshot();
+        assert_snapshot!(snapshot, @"ZQZQ :: Current module");
+    }
+
+    #[test]
+    fn reexport_simple_import_auto() {
+        let snapshot = CursorTest::builder()
+            .source(
+                "main.py",
+                r#"
+ZQ<CURSOR>
+"#,
+            )
+            .source("foo.py", r#"from bar import ZQZQ"#)
+            .source("bar.py", r#"ZQZQ = 1"#)
+            .completion_test_builder()
+            .auto_import()
+            .module_names()
+            .build()
+            .snapshot();
+        assert_snapshot!(snapshot, @"ZQZQ :: bar");
```

Comment:
I guess the real test here is that (unlike in the case below with the explicit redundant-alias re-export convention), `"ZQZQ :: foo"` does _not_ show up here. This definitely confused me on my first read of this test -- "Why is `foo.py` even here? How is this a test of re-exports? What are we even testing for here?". I generally find that some comments regarding intent can help clarify such negative tests a lot. (Or really all tests -- but I'm not trying to dictate your test-writing style here, if you generally don't comment them.)

Reply:
Yeah I can add a comment here. I sometimes comment tests. I can at least go through and add a quick note on tests expecting a negative result.